postinstall.js

```diff
@@ -71,8 +72,17 @@ function hasCache() {
 function download(cb) {
 	console.log('ngrok - downloading binary ' + cdnUrl);
 
+	const caString = execSync('npm config get ca', {stdio: ['ignore', 'pipe', 'pipe']})
+		.toString()
+		.replace(/\'/g, '\"')
+	const ca = caString.substring(0, 25) === '[ "-----BEGIN CERTIFICATE' ? JSON.parse(caString)[0] : undefined;
+	const options = {
+		url: cdnUrl,
+		ca
+	};
+
 	const downloadStream = request
-		.get(cdnUrl)
+		.get(options)
 		.on('response', res => {
 			if (!/2\d\d/.test(res.statusCode)) {
 				res.pause();
```

Comment:
please add a little integration test for postinstall.js to
- place certificate
- call postinstall
- verify that certificate was respected and nothing crashed :)

Reply:
Assuming the environment variable doesn't work, I will.

Reply:
Doing this will add another dev dependency (for the cert generation bit). I will add it for now. I can revert that commit if you'd rather not add the dependency.

Reply:
I've added this, but my generated root cert is probably bad. Being inside the corporate network, I can't test the cert, so I will have to finish this when I get home.

Reply:
Updated. If there is junk data, the script will just work. If there is a real cert that isn't the root CA, it's expected that the postinstall will fail. I added a test for both cases.